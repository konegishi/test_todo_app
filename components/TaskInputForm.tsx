import React, { useState } from 'react';
import Link from 'next/link';

/**
 * TaskInputFormのProps
 */
interface TaskInputFormProps {
  /** タスク名 */
  name?: string;
}

/**
 * TaskInputFormコンポーネント
 * タスクを入力するコンポーネント
 *
 * @param props TaskInputFormのProps
 * @returns TaskInputFormコンポーネント
 */
const TaskInputForm: React.FC<TaskInputFormProps> = () => {
  //   const [isOpen, setOpen] = useState(false);
  const [collapseShow, setCollapseShow] = useState('hidden');

  return (
    <React.Fragment>
      <form className='bg-white rounded-t-xl shadow-lg p-2 flex flex-row'>
        <input
          className='appearance-none border rounded-l py-2 px-3 w-11/12 text-gray-700 border-gray-200 leading-tight focus:outline-none focus:shadow-outline focus:border-blue-500'
          id='taskTitle'
          type='text'
          placeholder='新しいタスクの名前'
        />
        <button
          className='fas fa-plus flex-grow-0 bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded-r focus:outline-none focus:shadow-outline'
          type='button'
          onClick={() => setCollapseShow('bg-white m-2 py-3 px-6')}
          //   onClick={() => setOpen(!isOpen)}
        />
      </form>

      {/* Collapse */}
      <div
        className={
          'md:flex md:flex-col md:items-stretch md:opacity-100 md:relative md:mt-4 md:shadow-none shadow absolute bottom-0 left-0 right-0 z-40 overflow-y-auto overflow-x-hidden h-auto items-center flex-1 rounded ' +
          collapseShow
        }
      >
        {/* Collapse header */}
        <div className='md:min-w-full md:hidden block pb-3 mb-4 border-b border-solid border-blueGray-200'>
          <div className='flex flex-wrap'>
            <div className='w-6/12'>
              <span className='md:block text-left md:pb-2 text-blueGray-600 mr-0 inline-block whitespace-nowrap text-sm font-bold px-0'>
                Ape PEX
              </span>
            </div>
            <div className='w-6/12 flex justify-end'>
              <button
                type='button'
                className='cursor-pointer text-black opacity-50 md:hidden px-3 py-1 text-xl leading-none bg-transparent rounded border border-solid border-transparent'
                onClick={() => setCollapseShow('hidden')}
              >
                <i className='fas fa-times'></i>
              </button>
            </div>
          </div>
        </div>
        {/* Navigation */}
        <ul className='md:flex-col md:min-w-full flex flex-col list-none'>
          <li className='items-center'>
            <Link href='/dashboard'>
              <a className='text-pink-500 hover:text-pink-600 text-xs uppercase py-3 font-bold block'>
                <i className='fas fa-tv opacity-75 mr-2 text-sm'></i> Dashboard
              </a>
            </Link>
          </li>

          <li className='items-center'>
            <Link href='/'>
              <a className='text-blueGray-700 hover:text-blueGray-500 text-xs uppercase py-3 font-bold block'>
                <i className='fas fa-tv opacity-75 mr-2 text-sm'></i> Landing
                Page
              </a>
            </Link>
          </li>

          <li className='items-center'>
            <Link href='/'>
              <a className='text-blueGray-700 hover:text-blueGray-500 text-xs uppercase py-3 font-bold block'>
                <i className='fas fa-user-circle text-blueGray-400 mr-2 text-sm'></i>{' '}
                Profile Page
              </a>
            </Link>
          </li>

          <li className='items-center'>
            <Link href='/'>
              <a className='text-blueGray-700 hover:text-blueGray-500 text-xs uppercase py-3 font-bold block'>
                <i className='fas fa-fingerprint text-blueGray-400 mr-2 text-sm'></i>{' '}
                Login
              </a>
            </Link>
          </li>

          <li className='items-center'>
            <a
              className='text-blueGray-300 text-xs uppercase py-3 font-bold block'
              href='#pablo'
              onClick={(e) => e.preventDefault()}
            >
              <i className='fas fa-clipboard-list text-blueGray-300 mr-2 text-sm'></i>{' '}
              Register (soon)
            </a>
          </li>

          <li className='items-center'>
            <button>ログアウト</button>
          </li>
        </ul>
      </div>
      {/* <div
        className={`md:flex 
          ${
            !isOpen
              ? 'hidden'
              : 'bg-white absolute left-0 right-0 z-40 px-6 pt-4 overflow-y-auto overflow-x-hidden h-auto rounded'
          }`}
      >
        <div className='flex flex-col md:flex-row'>
          <span>hoge</span>
        </div>
      </div> */}
    </React.Fragment>
  );
};

export default TaskInputForm;
