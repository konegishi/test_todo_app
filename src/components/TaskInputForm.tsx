import { User } from '@supabase/supabase-js';
import React, { useRef, useState } from 'react';

/**
 * TaskInputFormのProps
 */
interface TaskInputFormProps {
  /** ユーザー情報 */
  user: User;
  /** タスク名 */
  name?: string;
  /** AddボタンのHandler */
  addTodoHandler: (user: User, taskText: string) => Promise<void>;
}

/**
 * TaskInputFormコンポーネント
 * タスクを入力するコンポーネント
 *
 * @param props TaskInputFormのProps
 * @returns TaskInputFormコンポーネント
 */
const TaskInputForm: React.FC<TaskInputFormProps> = (props) => {
  const [isOpen, setOpen] = useState(false);
  const inputRef = useRef<HTMLInputElement>(null);

  /**
   * ファームのタスク追加ボタンのhandler
   */
  const handleOnClickAddButton = () => {
    props.addTodoHandler(props.user, inputRef.current.value);
    inputRef.current.value = '';
    setOpen(!isOpen);
  };

  return isOpen ? (
    <React.Fragment>
      {/* blackOverlay */}
      <div className='absolute w-full min-h-full'>
        <span
          id='blackOverlay'
          className='absolute w-full min-h-full bg-gray-500 bg-opacity-75'
        ></span>
      </div>
      {/* タスク入力フォーム */}
      <div
        className={`${
          !isOpen
            ? 'hidden'
            : 'shadow sticky bottom-0 left-0 right-0 z-40 items-center rounded-t bg-white p-2'
        }`}
      >
        <div className='block mb-2'>
          <div className='flex flex-row items-center'>
            <div className='w-6/12'>
              <span className='text-left text-gray-700 mr-0 inline-block whitespace-nowrap text-sm font-bold px-0'>
                新しいタスクを入力してみましょう
              </span>
            </div>
            <div className='w-6/12 flex justify-end'>
              <button
                type='button'
                className='cursor-pointer text-black opacity-50 px-3 py-1 text-xl leading-none bg-transparent rounded border border-solid border-transparent'
                onClick={() => setOpen(!isOpen)}
              >
                <i className='fas fa-times'></i>
              </button>
            </div>
          </div>
        </div>
        {/* Form */}
        <form className='flex flex-row'>
          <input
            className='rounded-l py-2 px-3 w-11/12 text-gray-700 border-gray-200 leading-tight'
            id='taskTitle'
            ref={inputRef}
            type='text'
            placeholder='新しいタスクの名前'
          />
          <button
            className='fas fa-arrow-circle-up flex-grow-0 bg-blue-500 text-white font-bold py-2 px-4 rounded-r'
            type='button'
            onClick={() => handleOnClickAddButton()}
          />
        </form>
      </div>
    </React.Fragment>
  ) : (
    <React.Fragment>
      {/* FAB */}
      <div className='sticky mx-auto bottom-0'>
        <button
          className='fas fa-plus mb-2 rounded-full h-12 w-12 bg-blue-500 text-white text-xl'
          type='button'
          onClick={() => setOpen(!isOpen)}
        />
      </div>
    </React.Fragment>
  );
};

export default TaskInputForm;
