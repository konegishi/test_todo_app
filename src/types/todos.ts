import { User } from '@supabase/supabase-js';

/**
 * Todoの型定義
 */
export interface Todo {
  /** todo名 */
  task: string;
  /** todoのid */
  id: number;
  /** 説明 */
  description: string;
  /** 完了フラグ */
  isComplete: boolean;
}

export interface UseTodos {
  /** todoのリスト */
  todos: Todo[];
  /** ローディング中フラグ */
  isLoading: boolean;
  /**addTodoメソッドの型定義 */
  addTodo: (user: User, taskText: string) => Promise<void>;
  /** updateTodoメソッドの型定義 */
  updateTodo: (
    id: number,
    todo: {
      task?: string;
      description?: string;
      isComplete?: boolean;
    }
  ) => Promise<void>;
  /** deleteTodoメソッドの型定義 */
  deleteTodo: (id: number) => Promise<void>;
}
