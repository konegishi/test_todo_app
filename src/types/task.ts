import { UseTodos } from './todos';

/**
 * TaskコンポーネントのProps
 */
export interface TaskProps {
  /** タスク名 */
  title: string;
  /** タスクのid */
  id: number;
  /** タスクの説明 */
  description: string;
  /** タスクの完了フラグ */
  isComplete: boolean;
  /** DeleteボタンのHandler */
  deleteTodoHandler: UseTodos['deleteTodo'];
  /** UpdateボタンのHandler */
  updateTodoHandler: UseTodos['updateTodo'];
}
