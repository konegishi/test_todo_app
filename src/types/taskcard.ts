import { Todo, UseTodos } from './todos';

/**
 * TaskCardコンポーネントのProps
 */
export interface TaskCardProps {
  /** タスクのリスト */
  todos: Todo[];
  /** DeleteボタンのHandler */
  deleteTodoHandler: UseTodos['deleteTodo'];
  /** UpdateボタンのHandler */
  updateTodoHandler: UseTodos['updateTodo'];
}
