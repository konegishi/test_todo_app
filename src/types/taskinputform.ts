import { User } from '@supabase/supabase-js';
import { UseTodos } from './todos';

/**
 * TaskInputFormコンポーネントのProps
 */
export interface TaskInputFormProps {
  /** ユーザー情報 */
  user: User;
  /** タスク名 */
  name?: string;
  /** AddボタンのHandler */
  addTodoHandler: UseTodos['addTodo'];
}
