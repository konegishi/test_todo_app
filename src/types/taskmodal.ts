import { Dispatch, SetStateAction } from 'react';
import { UseTodos } from './todos';

/**
 * TaskModalコンポーネントのProps
 */
export interface TaskModalProps {
  /** タスク名 */
  title: string;
  /** タスクのid */
  id: number;
  /** タスクの説明 */
  description: string;
  /** Modal開閉用メソッド */
  onClickHandler: Dispatch<SetStateAction<boolean>>;
  /** DeleteボタンのHandler */
  deleteTodoHandler: UseTodos['deleteTodo'];
  /** UpdateボタンのHandler */
  updateTodoHandler: UseTodos['updateTodo'];
}
