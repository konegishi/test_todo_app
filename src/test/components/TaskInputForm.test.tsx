import { render } from '@testing-library/react';
import React from 'react';
import TaskInputForm from '../../components/TaskInputForm';
import { TaskInputFormProps } from '../../types/taskinputform';

describe('Task component', () => {
  it('matches snapshot', () => {
    const dummyProps: TaskInputFormProps = {
      user: {
        id: 'id',
        app_metadata: {
          provider: 'email',
        },
        user_metadata: {},
        aud: 'aud',
        created_at: '2021-11-01',
      },
      name: '型定義を追加する',
      addTodoHandler: jest.fn(),
    };
    const { asFragment } = render(<TaskInputForm {...dummyProps} />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
