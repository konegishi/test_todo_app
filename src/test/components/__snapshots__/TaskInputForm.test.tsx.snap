// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sticky mx-auto bottom-0"
  >
    <button
      class="fas fa-plus mb-2 rounded-full h-12 w-12 bg-blue-500 text-white text-xl"
      type="button"
    />
  </div>
</DocumentFragment>
`;
